lighthouse/environment/src/lib.rs

```diff
@@ -182,149 +175,59 @@ impl<E: EthSpec> EnvironmentBuilder<E> {
+        let mut path = PathBuf::new();
+        for p in log_path.iter() {
+            path = path.join(p);
+            if let Ok(metadata) = path.metadata() {
+                if !metadata.is_dir() {
+                    path.pop();
+                    break;
                 }
-                _ => return Err("Logging format provided is not supported".to_string()),
-            }
-        } else {
-            let stdout_decorator_builder = slog_term::TermDecorator::new();
-            let stdout_decorator = if config.log_color {
-                stdout_decorator_builder.force_color()
-            } else {
-                stdout_decorator_builder
             }
-            .build();
-            let stdout_decorator =
-                logging::AlignedTermDecorator::new(stdout_decorator, logging::MAX_MESSAGE_WIDTH);
-            let stdout_drain = slog_term::FullFormat::new(stdout_decorator);
-            let stdout_drain = if config.disable_log_timestamp {
-                stdout_drain.use_custom_timestamp(Self::log_nothing)
-            } else {
-                stdout_drain
-            }
-            .build()
-            .fuse();
-            slog_async::Async::new(stdout_drain)
-                .chan_size(LOG_CHANNEL_SIZE)
-                .build()
-        };
-
-        let stdout_drain = match config.debug_level.as_str() {
-            "info" => stdout_drain.filter_level(Level::Info),
-            "debug" => stdout_drain.filter_level(Level::Debug),
-            "trace" => stdout_drain.filter_level(Level::Trace),
-            "warn" => stdout_drain.filter_level(Level::Warning),
-            "error" => stdout_drain.filter_level(Level::Error),
-            "crit" => stdout_drain.filter_level(Level::Critical),
-            unknown => return Err(format!("Unknown debug-level: {}", unknown)),
+        }
+        let Ok(file_appender) = RollingFileAppender::builder()
+            .rotation(Rotation::DAILY)
+            .max_log_files(config.max_log_number)
+            .filename_prefix("beacon")
+            .filename_suffix("log")
+            .build(path.clone())
+        else {
+            panic!("Failed to initialize rolling file appender");
         };
 
-        let stdout_logger = Logger::root(stdout_drain.fuse(), o!());
-
-        // Disable file logging if values set to 0.
-        if config.max_log_size == 0 || config.max_log_number == 0 {
-            self.log = Some(stdout_logger);
-            return Ok(self);
-        }
+        let (file_non_blocking_writer, file_guard) = tracing_appender::non_blocking(file_appender);
 
-        // Disable file logging if no path is specified.
-        let Some(path) = config.path else {
-            self.log = Some(stdout_logger);
-            return Ok(self);
+        let file_logging_layer = LoggingLayer {
+            non_blocking_writer: file_non_blocking_writer,
+            guard: file_guard,
+            disable_log_timestamp: config.disable_log_timestamp,
         };
```

Comment:
Ideally we need to preserve this feature of being able to disable file logging. Perhaps we can only build the `RollingFileAppender` if `path` is set in the config?